lib/client/db/mysql/repl/repl.go

```diff
@@ -0,0 +1,271 @@
+			if exit {
+				return nil
+			}
+			if err := r.print(reply); err != nil {
+				return trace.Wrap(err)
+			}
+		}
+	}
+}
+
+func withClientCapabilities(caps ...uint32) client.Option {
+	return func(conn *client.Conn) error {
+		for _, cap := range caps {
+			conn.SetCapability(cap)
+		}
+		return nil
+	}
+}
+
+func (r *REPL) presentBanner() error {
+	_, err := fmt.Fprintf(
+		r.term,
+		`Teleport MySQL interactive shell (v%s)
+Connected to instance %q as user %q.
+Type 'help' or '\h' for help.
+
+`,
+		cmp.Or(r.teleportVersion, teleport.Version),
+		r.route.GetServiceName(),
+		r.route.GetUsername())
+	return trace.Wrap(err)
+}
+
+func (r *REPL) readLine() (string, error) {
+	r.term.SetPrompt(r.getPrompt())
+	return r.term.ReadLine()
+}
+
+func (r *REPL) getPrompt() string {
+	dbName := formatDatabaseName(r.route.Database)
+	var suffix string
+	switch {
+	case r.parser.lex.isInComment():
+		suffix = "/*"
+	case r.parser.lex.isInString():
+		suffix = r.parser.lex.inStringKind()
+	case r.parser.lex.isInQuery():
+		suffix = "-"
+	default:
+		suffix = dbName
+	}
+	padding := len(dbName)
+	// pad the suffix with leading spaces until it is len(dbName), for example:
+	// default: "mysql>"
+	// comment: "   */>"
+	// string:  "    '>"
+	// query:   "    ->"
+	return fmt.Sprintf("%*s> ", padding, suffix)
+}
+
+func (r *REPL) eval(line string) iter.Seq2[string, bool] {
+	return func(yield func(string, bool) bool) {
+		for evaluator := range r.parser.parse(line) {
+			if !yield(evaluator.eval(r)) {
+				return
+			}
+		}
+	}
+}
+
+// print writes a reply to the REPL client.
+func (r *REPL) print(reply string) error {
+	reply = strings.TrimSpace(reply)
+	if reply == "" {
+		return nil
+	}
+	_, err := r.term.Write([]byte(lineBreak + reply + lineBreak + lineBreak))
+	return trace.Wrap(err)
+}
+
+// rewriteTermError changes the term.Terminal error to match caller expectations.
+func rewriteTermError(ctx context.Context, err error) error {
+	// When context is canceled it will immediately lead read/write errors due
+	// to the closed connection. For this cases we return the context error.
+	if ctx.Err() != nil && (errors.Is(err, io.ErrClosedPipe) || errors.Is(err, net.ErrClosed)) {
+		return ctx.Err()
+	}
+	if errors.Is(err, io.EOF) {
+		return nil
+	}
+
+	return err
+}
+
+// formatResult formats a query result.
+func formatResult(result *mysql.Result) string {
+	if result.Resultset != nil {
+		return lineBreak + formatResultset(result.Resultset)
+	}
+	return lineBreak + fmt.Sprintf("%v warnings, %v rows affected", result.Warnings, result.AffectedRows)
+}
+
+// formatResultset formats a query result that contains rows.
+func formatResultset(set *mysql.Resultset) string {
+	if set == nil || len(set.Values) == 0 {
+		return "Empty set"
+	}
+	var sb strings.Builder
+	columns := make([]string, 0, len(set.Fields))
+	for _, f := range set.Fields {
+		columns = append(columns, string(f.Name))
+	}
+	table := asciitable.MakeTable(columns)
```

Comment:
minor UX improvements, if not a lot of work:
```
mysql> select sleep(5);
+----------+
| sleep(5) |
+----------+
|        0 |
+----------+
1 row in set (5.009 sec)
```

- is it possible for our table to print the border similar to `mysql` client?
- could we also add the duration?